### DISCUSSION TITLE
How to query issues that match a text in their title

### DISCUSSION THREAD

**Discussion by hosain-cse2** [community/community#24428] [Apps, API and Webhooks]
<p>Hi,</p>
<p>I am new in graphql. I want to make a query or search to get all the issues that matches a text in the title. I could not find any solution or may be I did not understand the solution since I do not have much idea about graphql or GitHub graphql API.</p>
<p>Can please tell how will be the query if I want to get all the issues, suppose from react GitHub repository, where all the title of the issues has text e.g. “error”</p>
<p>Thanks.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>The <a href="https://developer.github.com/v4/query/#connections" rel="nofollow noopener">search connection</a> query parameter uses <a href="https://help.github.com/en/categories/searching-for-information-on-github" rel="nofollow noopener">the same syntax as the search box on github.com</a>. Therefore this should work for you:</p>
<pre><code>{
  search(query: "repo:lee-dohm/test-repo in:title test", type: ISSUE, first: 100) {
    nodes {
      ... on Issue {
        number
        title
        body
      }
    }
  }
}
</code></pre>
<p>I hope that helps!</p>

**Reply by rafecolton** [reply]
This is great, thanks! Do you happen to know if the search api rate limit applies to search over graphql as well? 

**Edit:** I confirmed with GH support, the rate limits are separate.

**Reply by ndiritumichael** [reply]
hi this doesn't work anymore with the graphql api

**Reply by rafecolton** [reply]
@ndiritumichael what specific error are you seeing? Works fine for me...

**Comment by cchung85** [comment]
Here is a simplified answer, partly based on @lee-dohm reply:
Say you are in github.com tensorflow/datasets, lets say you want to search for issues with key word imagenet in the title:

In the Filters box (towards the top of the window):
1. All open issue with imagenet in title
is:issue   is:open  in:title imagenet
2. All issues, open or closed, with imagenet in the title:
is:issue    in:title imagenet
3. All open issues with imagenet in the text:
is:issue   is:open  in:text imagenet

**Reply by DeepikaRani-Hub** [reply]
hi,

how to pass  "test" as a variable to the query. i modified this query by adding test as dynamic data but getting some errors.

query:

query SearchIssues($text: String!, $status: String!) {
    search(query: "repo:facebook/react in:[title or body] $text is:$status", type: ISSUE, first: 100) {
      nodes {
        ... on Issue {
          number
          title
          body
        }
      }
    }
  }

query variables:

{
  "text": "test",
  "status": "open"
}

error:

"message": "Variable $text, $status is declared by SearchIssues but not used"
